src/Storages/ObjectStorage/StorageObjectStorageSource.cpp

```diff
@@ -470,21 +496,7 @@ StorageObjectStorageSource::ReaderHolder StorageObjectStorageSource::createReade
         if (object_info->getPath().empty())
             return {};
 
-        if (!object_info->metadata)
-        {
-            const auto & path = object_info->isArchive() ? object_info->getPathToArchive() : object_info->getPath();
-
-            if (query_settings.ignore_non_existent_file)
-            {
-                auto metadata = object_storage->tryGetObjectMetadata(path);
-                if (!metadata)
-                    return {};
-
-                object_info->metadata = metadata;
-            }
-            else
-                object_info->metadata = object_storage->getObjectMetadata(path);
-        }
+        object_info->loadMetadata(object_storage, query_settings.ignore_non_existent_file);
```

Comment:
In the oroginal PRs it was just `object_info->loadMetadata(object_storage)`, but signature was changed, used `query_settings.ignore_non_existent_file` to plug the hole. Again, not sure if that's the right thing to do (@arthurpassos please check, and other instances of calling `object_info->loadMetadata` too)